sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -259,4 +259,48 @@ object ParserUtils {
       }
     }
   }
+
+  private def stripGap(sb: StringBuilder): Unit = {
+    if (sb.length > 0 && sb.charAt(sb.length - 1) == ' ') {
+      sb.setLength(sb.length - 1)
+    }
+  }
+
+  /**
+   * Adds a gap after every term node except of '(', ')', '[', ']',
+   * and removes a gap before '(', ')', '[', ']', ','.
```

Comment:
can we give a few examples to help people understand the rule?

Reply:
added some examples